File: 백준/Silver/14916. 거스름돈/거스름돈.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int coin = Integer.parseInt(br.readLine());
        int cnt = 0;
        while (coin > 0) {
            if (coin % 5 == 0) {
                cnt += coin / 5;
                break;
            } else {
                coin -= 2;
                cnt++;
            }
        }
        if (coin < 0) {
            System.out.println(-1);
        } else {
            System.out.println(cnt);
        }
    }
}

File: 백준/Silver/14916. 거스름돈/거스름돈.py
import sys

input = sys.stdin.readline

n = int(input())
coin = 0

while True:
    if n % 5 == 0:
        coin += n // 5
        break

    if n == 2:
        coin += 1
        break

    n -= 2
    coin += 1

    if n <= 1:
        print(-1)
        exit()
print(coin)